File: top-interview-questions-easy/Array/move-zeroes/283.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
  var len = nums.length
  var i, j
  for (i = 0; i < len - 1; i++) {
    if (nums[i]) continue
    for (j = i + 1; j < len; j++) {
      if (!nums[j]) continue
      [nums[i], nums[j]] = [nums[j], nums[i]]
      break
    }
  }
}

File: 283.移动零/283.py
from typing import List


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        for i in range(len(nums))[::-1]:
            if (nums[i]):
                continue
            nums.append(nums.pop(i))


list = [0, 1, 0, 3, 12]
s = Solution()
s.moveZeroes(list)
print(list)
